### PR TITLE
Fix override_settings issue

### PR DESCRIPTION
**Pull request by ewjoachim**
See #203 for context.

Also, this is enough to fix the issue in the reproduction repo.

### CODE REVIEW

**Comment by codingjoe** [comment]
I will close this until we have reached a conses in the issue, OK?

**Comment by ewjoachim** [comment]
> You can add Django unit tests to pytest

I don't need to, I'm pretty sure it happens identically with Pytest. The only reason I didn't use pytest on the repro repo was to keep things as simple as possible :)

File: django_select2/apps.py [hunk 1]
```diff
@@ -7,3 +7,6 @@ class Select2AppConfig(AppConfig):
 
     name = "django_select2"
     verbose_name = "Select2"
+
+    def ready(self):
+        from . import conf
```
[the remark this answers is not in the document]
Comment (ewjoachim) [reply django_select2/apps.py]: (there's an extra `.` in your suggestion, I'll fix it too)

**Comment by ewjoachim** [comment]
Ok, if you really want to have a test on this we're going to need black magic or a very specific setup.

In the tests, as they currently exist, the conf module is loaded as a side effect of importing other modules, which happens during test collection, before tests actually start running.
The only ways I see to reproduce the issue would be:
- To have our test run before other tests are collected. The only way to achieve this is to run this test in a dedicated pytest process. That's a very specific setup, it means we can't run all the test just by launching pytest, we'd need to launch both pytest on 1 subfolder and then pytest on everything
- Or to force Python to ignore that `conf` has already been loaded, e.g. by using `reload()` or deleting stuff from `sys.modules` manually. I'm calling this black magic, and as you know, all magic comes with a price (and I'm not sure I'm ready to pay the price)

If you have another idea, I'm all ears.

**Review by codingjoe — APPROVED** [review]

**Comment by ewjoachim** [comment]
Thanks

**Pull request merged**